### PR TITLE
JoinSubtitles: Update

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

File: src/Forms/JoinSubtitles.cs [hunk 1]
```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
...
 
-                    if (lastFormat == null || lastFormat.FriendlyName == format.FriendlyName)
+                    // set last-format if not set yet
+                    if (lastFormat == null)
                     {
                         lastFormat = format;
                     }
                     else
                     {
-                        lastFormat = new SubRip(); // default subtitle format
```
Comment (ivandrofly) [inline src/Forms/JoinSubtitles.cs]: Unnecessary instance of subrip were being created every time previous format isn't same as current format in loop

File: src/Forms/JoinSubtitles.cs [hunk 2]
```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
...
+                        {
+                            lastFormat = OutputFormat;
+                        }
                     }
 
                     subtitles.Add(sub);
                 }
                 catch (Exception exception)
                 {
-                    for (int j = k; j < _fileNamesToJoin.Count; j++)
```
Comment (ivandrofly) [inline src/Forms/JoinSubtitles.cs]: duplicated code

File: src/Forms/Main.cs [hunk 3]
```diff
@@ -23754,25 +23754,32 @@ private void joinSubtitlesToolStripMenuItem_Click(object sender, EventArgs e)
...
-                    ResetSubtitle();
-                    _subtitle = joinSubtitles.JoinedSubtitle;
-                    SetCurrentFormat(joinSubtitles.JoinedFormat);
-                    SubtitleListview1.Fill(_subtitle, _subtitleAlternate);
-                    SubtitleListview1.SelectIndexAndEnsureVisible(0, true);
+                MakeHistoryForUndo(_language.BeforeDisplaySubtitleJoin);
 
-                    if (IsFramesRelevant && CurrentFrameRate > 0)
-                    {
-                        _subtitle.CalculateFrameNumbersFromTimeCodesNoCheck(CurrentFrameRate);
```
Comment (ivandrofly) [inline src/Forms/Main.cs]: subtitle.loadsubtitle already handles frame-based formats

File: src/Forms/JoinSubtitles.cs [hunk 4]
```diff
@@ -250,15 +226,7 @@ private void ButtonAddSubtitleClick(object sender, EventArgs e)
...
                         {
-                            bool alreadyInList = false;
-                            foreach (string existingFileName in _fileNamesToJoin)
-                            {
-                                if (existingFileName.Equals(fileName, StringComparison.OrdinalIgnoreCase))
-                                {
-                                    alreadyInList = true;
-                                }
-                            }
-                            if (!alreadyInList)
```
Comment (ivandrofly) [inline src/Forms/JoinSubtitles.cs]: replaced with linq.any

File: src/Forms/JoinSubtitles.cs [hunk 5]
```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
...
-                        _fileNamesToJoin[inner - 1] = _fileNamesToJoin[inner];
-                        _fileNamesToJoin[inner] = t1;
+            OutputFormat = lastFormat;
 
-                        var t2 = subtitles[inner - 1];
-                        subtitles[inner - 1] = subtitles[inner];
-                        subtitles[inner] = t2;
-                    }
-                }
-            }
```
Comment (ivandrofly) [inline src/Forms/JoinSubtitles.cs]: replaced w/ linq with is more readable and which also made it easier to generate view

File: src/Forms/JoinSubtitles.cs [hunk 6]
```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
...
-                    }
-
-                    if (format == null)
-                    {
-                        for (int j = k; j < _fileNamesToJoin.Count; j++)
-                        {
-                            _fileNamesToJoin.RemoveAt(j);
-                        }
-
-                        MessageBox.Show("Unkown subtitle format: " + fileName);
```
Comment (ivandrofly) [inline src/Forms/JoinSubtitles.cs]: duplicated codes (all moved to a method)

**Comment by niksedk** [comment]
Seems buggy... I got double number of lines second time I tried it.

**Comment by niksedk** [comment]
Perhaps you could split it up in minor parts? Linq + binary loading looks good :)

**Pull request closed without merging**